design/animation.md

```diff
@@ -0,0 +1,360 @@
+Animations
+===========
+
+Our current animation system is under-featured, unstable, and based on tech (Unity Animation components) that is too
+rigid for our purposes. We need an animation system that can support the following:
+
+1. Native animations - Animations added to actors from glTF or native Unity sources should support the basic API: play,
+	pause, seek.
+2. Keyframes - Vary actor/asset fields through a sequence of values on a timeline.
+	1. Interpolation - Make simple single-target two-frame animations easy to generate.
+	2. Moving targets - Setting/interpolating to/from another actor's dynamic values. Useful for look-at and
+		fly-to type animations.
+	3. Relative values - Add keyframe value to existing value instead of replacing it.
+3. Easing - Support different interpolation curves between keyframes.
+	1. No easing - Setting fields that cannot be interpolated should also be supported (string, boolean, etc.).
+4. Chaining - Begin an animation immediately after another finishes, or on some other event (behavior tie-in?)
+5. Multiple targets - Animate multiple actors/assets, and multiple properties, in lock-step (equivalent to Unity's
+	animation curves)
+6. Blending - Allow multiple animations to run on the same target simultaneously, and blend the computed destinations
+	into one, proportional to a weight property.
+
+To address this need, I propose the following API:
+
+API Design
```

Comment:
Goals: I think we should remove "Chaining" add a more generalized goal: "Control". Support starting and stopping animations together, and trigger animations from other animations.

Reply:
From chatting offline: Steven's going to explore the idea of making `AnimationInstance` itself animatable. That way we could start and stop animations from other animations the same way we will be animating actor and materials.

Reply:
... and this would make `continueWith` unnecessary.